### PR TITLE
[DEV-9503] Add opt-in support for pinned stories

### PR DESCRIPTION
**Pull request by e1himself**
Depends on this: :heavy_check_mark: https://github.com/prezly/javascript-sdk/pull/186

### CODE REVIEW

File: src/data-fetching/api/PrezlyApi.ts [hunk 1]
```diff
@@ -108,8 +107,8 @@ export class PrezlyApi {
     /**
      * Note: this method returns ALL stories from the newsroom. It's intended to be used for sitemaps and not to display actual content.
      */
-    async getAllStories(order: SortOrder = DEFAULT_SORT_ORDER) {
-        const sortOrder = getSortByPublishedDate(order);
+    async getAllStories({ order = DEFAULT_SORT_ORDER, pinning = false }: GetStoriesOptions = {}) {
```
Comment (e1himself) [inline src/data-fetching/api/PrezlyApi.ts:110]: This is a breaking change. Will bump major version.

**Review by fgyimah — APPROVED** [review]
👍🏾

File: src/adapter-nextjs/page-props/sitemap/getServerSideProps.ts [hunk 2]
```diff
@@ -5,7 +5,9 @@ import { getPrezlyApi } from '../../../data-fetching';
 import { createPaths } from './createPaths';
 import { SitemapBuilder } from './SitemapBuilder';
 
-export function getSitemapServerSideProps(additionalPaths?: string[]) {
+export function getSitemapServerSideProps(
+    options: { additionalPaths?: string[]; pinning?: boolean } = {},
+) {
```
[the remark this answers is not in the document]
Comment (aspirisen) [reply src/adapter-nextjs/page-props/sitemap/getServerSideProps.ts:10]: @e1himself am i right that major versions are not updated by `renovate`? If yes, are you going to update theme-kit in all themes?

[the remark this answers is not in the document]
Comment (aspirisen) [reply src/adapter-nextjs/page-props/sitemap/getServerSideProps.ts:10]: @oleg-semyonov do you know if `renovate` updates major verions?

@e1himself if renovate doesn't update major versions then we can face the situation where the themes will have outdated libraries and when you will need to bring feature to all of them you will have to make version upgrade as well, everywhere, at once

[the remark this answers is not in the document]
Comment (e1himself) [reply src/adapter-nextjs/page-props/sitemap/getServerSideProps.ts:10]: > when you will need to bring feature to all of them you will have to make version upgrade as well, everywhere, simultaneously

Isn't it how things are usually done? The next person who requires an updated version of the package deals with the upgrade.

Like here, I had to address `seo_settings` property (introduced by a previous release of `@prezly/sdk`).

[the remark this answers is not in the document]
Comment (oleg-semyonov) [reply src/adapter-nextjs/page-props/sitemap/getServerSideProps.ts:10]: Renovate does not update major versions, only minor and patches. Major version updates should be approved manually through the special issue on the repo

**Comment by oleg-semyonov** [comment]
Honestly, I don't like this approach where we deliberately blocking out some themes from the feature, when all it changes is the order of the stories in the array.
I posted my thoughts on this (and so did Istvan) in [this Linear ticket](https://linear.app/prezly/issue/DEV-9503#comment-b34ca04a)

File: src/data-fetching/api/PrezlyApi.ts (hunk 1 above)
Comment (oleg-semyonov) [reply src/data-fetching/api/PrezlyApi.ts:110]: I don't see how this is a breaking change 🤔 
`pinning` parameter is optional, and without it, nothing would change, right?

Comment (e1himself) [reply src/data-fetching/api/PrezlyApi.ts:110]: It *is* a breaking change. There was a single positional argument of string type. Now it's a destructured object.

**Comment by e1himself** [comment]
@oleg-semyonov 

> Honestly, I don't like this approach where we're deliberately blocking out some themes from the feature, when all it changes is the order of the stories in the array.

Alright. I don't fully buy it either. But do you approve this change or now? :)

File: src/data-fetching/api/PrezlyApi.ts (hunk 1 above)
Comment (oleg-semyonov) [reply src/data-fetching/api/PrezlyApi.ts:110]: Oh right, I missed it! Now it makes sense.

**Review by oleg-semyonov — APPROVED** [review]
I'm approving this to unblock your project work 👍🏻 
I would still be happy to see the `pinning` option default to `true`, if that's reasonable. I've just got a feeling that we'll have to remove this option afterwards anyway.

**Comment by aspirisen** [comment]
Personally i do not like the idea to leave other themes outdated, when you will release major release then other themes will be harder to updated in future. While we will be updating theme kit other themes will have the old version and even if we will need to add some small feature to themes we will have to make huge upgrade of every theme

**Comment by e1himself** [comment]
> Personally i do not like the idea to leave other themes outdated, when you will release major release then other themes will be harder to updated in future.

Yeah, that sucks. But upgrading all themes at once with every major update of this library is even worse. So it's just about picking the least evil of two, I believe.

**Comment by aspirisen** [comment]
Well, i will have to add new meta tag in all themes in scope of current project. Now i will have to upgrade all the themes and fix the errors that i am not belong to

**Comment by oleg-semyonov** [comment]
@aspirisen You won't need to do anything besides bumping the package, as pinning is opt-in. We don't use custom sitemap argument, so the breaking change won't affect our themes

**Pull request merged**